src/home/homework.c

```diff
@@ -1795,8 +1799,43 @@ static int home_inspect(UserRecord *h, UserRecord **ret_home) {
         return 1;
 }
 
+static int user_session_freezer(UserRecord *h, bool freeze, UnitFreezer *ret) {
+        _cleanup_free_ char *unit = NULL;
+        int should_freeze = h->freeze_session;
+
+        if (should_freeze < 0) {
+                should_freeze = getenv_bool("SYSTEMD_HOME_FREEZE_SESSION_OVERRIDE_DEFAULT");
+                if (should_freeze < 0) {
+                        if (should_freeze != -ENXIO)
+                                log_warning_errno(should_freeze, "Failed to parse $SYSTEMD_HOME_FREEZE_SESSION_OVERRIDE_DEFAULT, ignoring.");
+                        should_freeze = TRUE;
+                }
+        }
+        if (!should_freeze)
+                return 0;
+
+        // TODO: Figure out if we should be freezing:
+        //   user@<UID>.service, OR
+        //   user-<UID>.slice
+        //
+        // Freezing the slice breaks GDM re-authentication, because it
+        // contains session-<ID>.scope, which contains gdm-session-worker.
+        // This is used by GDM to re-authenticate and unlock a session. Thus,
+        // if we freeze the whole user-<UID>.slice, there's no reauth worker.
+        // HOWEVER, I think conceptually it makes more sense to freeze the whole
+        // slice. Will need to discuss w/ GDM maintainers
```

Comment:
Reminder

Reply:
i think this is a non-issue. gdm needs some reworks anyway to allow suspended home dirs. for now pam_systemd_home.so for gdm will run with suspend=false, so that the home dirs are not suspended anyway.

Reply:
Sure it's a non-issue for systemd, I agree. It's an issue for me personally, who's going to be the one reworking GDM to properly support homed

However, I've already reached out to the maintainers of GDM and they pointed out where the problem probably is and what I can do about it. Hopefully I can get a fixup for this stalling in soon...